back/engines/commercial/insights/app/finders/insights/inputs_finder.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Insights
+  class InputsFinder
+
+    MAX_PER_PAGE = 100
+
+    attr_reader :view, :params
+
+    def initialize(view, params = {})
+      @view = view
+      @params = params
+    end
+
+    def execute
+      inputs = view.scope.ideas
+      inputs = filter_category(inputs)
+      inputs = search(inputs)
+      paginate(inputs)
+    end
+
+    # Takes into account, both, actual and suggested categories.
+    # Keep only inputs without categories if +params[:category]+ is +nil+ 
```

Comment:
I might be completely mistaken, but it seems a bit of a weird choice to get inputs with no categories with both '' and nil, do we do that anywhere else ? Could we consider the more explicit (to me) choice to only respond that for '' ?

Reply:
Initially, I was aiming for `nil` only, bc it feels a more natural way to do it in ruby. But I could not find a way to pass nil-null value for query parameter through the endpoint. So in the end, I opted for something like: `GET .../views/:view_id/inputs?category=` which assigns the empty string value to the category parameters.

I like to be able to use `nil` bc semantically it's very clean. On the other hand, it's nice to be able to pass the params directly to the finder 💔 

Another solution would be to convert the '' into nil when parsing/whitelisting the parameters in the controller. But this would make the parsing code a tiny bit more complicated.

Reply:
Thanks for this answer, clear !